src/Microsoft.FeatureManagement/FeatureManagementBuilderExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.FeatureManagement.Telemetry;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// Extensions used to add feature management functionality.
+    /// </summary>
+    public static class FeatureManagementBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a telemetry publisher to the feature management system.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFeatureManagementBuilder"/> used to customize feature management functionality.</param>
+        /// <returns>A <see cref="IFeatureManagementBuilder"/> that can be used to customize feature management functionality.</returns>
+        public static IFeatureManagementBuilder AddTelemetryPublisher<T>(this IFeatureManagementBuilder builder) where T : ITelemetryPublisher
```

Comment:
Should the API be simplified to `AddTelemetry` and it accepts a "publisher" for the telemetry?

For example, there is a `IConfigurationBuilder.Add<TSource>` instead of `IConfigurationBuilder.AddSource<TSource>`. We could have `AddTelemetry<TPublisher>` if it helps.

Reply:
Hmm, not sure. To me, it should make sense to run multiple times. I feel like AddTelemetryPublisher makes a little more sense in that regard, but both work.

Reply:
@zhenlan if we went with that pattern then it seems it would just be `Add<ITelemetryPublisher>`

So

`AddTelemetryPublisher<ITelemetryPublisher>`

or 

`Add<ITelemetryPublisher>`

?

Reply:
If telemetry is all we have, then `Add<ITelemetryPublisher>` would make sense. A standalone `Add` could be confusing since we have other `Add*` methods. Looks like you guys prefer the existing one. I think we should stick with that one, then.